django-stubs/contrib/admin/decorators.pyi

```diff
@@ -17,21 +17,21 @@ _QuerySet = TypeVar("_QuerySet", bound=QuerySet)
 # This is deliberately different from _DisplayT defined in contrib.admin.options
 _DisplayCallable: TypeAlias = Union[Callable[[_ModelAdmin, _Model], Any], Callable[[_Model], Any]]  # noqa: Y037
 _DisplayCallableT = TypeVar("_DisplayCallableT", bound=_DisplayCallable)
+_ActionCallable: TypeAlias = Callable[[_ModelAdmin, _Request, _QuerySet], HttpResponseBase | None]  # noqa: Y037
+_ActionCallableT = TypeVar("_ActionCallableT", bound=_ActionCallable)
```

Comment:
I think there was a good reason why these `Callable[...]` were duplicated before at each overload.

I'm not sure why exactly -- maybe mypy doesn't support TypeVars inside TypeVars -- but after your changes, this no longer raises an error with mypy at the location of the decorator:

```
    @admin.action(description="Some text here", permissions=["test"])
    def freestanding_action_fancy(modeladmin: int, request: int, queryset: int) -> None: ...
```

I suppose this logic could also use a few _negative_ test cases -- checking that mypy correctly flags incorrect usage of the parameters or return types.

Reply:
Yes, `TypeVar` can't be bound to a `TypeAlias`, it'll resolve to `Any` before it's matched with generic arguments of a class.

A few non-happy test cases should make that obvious.

Reply:
Sounds good, I've reverted the previous change and added more tests to highlight this issue. I've left `_DisplayCallableT` alone, but I'd imagine it probably has a similar problem.

Reply:
I see… Yes, if it's a problem, then the equivalent for display is also surely broken.

Reply:
I wonder whether it'd be worthwhile to enable `disallow_any_generics` for this repo -- I put together a quick repro, and it looks like if we enable the rule we'll get warnings for this: https://mypy-play.net/?mypy=latest&python=3.11&flags=disallow-any-generics&gist=bf198a38181d36c2105ce3fef468d702

Reply:
Yeah, maybe, but best looked into separately later.

Reply:
I'm afraid the fallout from `disallow_any_generics` will be enormous. But I may be wrong :)

Reply:
> Found 630 errors in 155 files (checked 675 source files)

You aren't wrong. 😬 Here's a quick stab at it though: https://github.com/typeddjango/django-stubs/pull/1339